trending_data/src/main/java/com/google/sps/servlets/Search.java

```diff
@@ -0,0 +1,107 @@
+import com.google.api.services.youtube.model.Video;
+import com.google.api.services.youtube.model.VideoListResponse;
+import com.google.sps.data.YTVid;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Properties;
+
+public class Search {
+
+  /** Identify the developer's API key. */
+  private static final String apiKey =
+      "AIzaSyClpjaOk1vj_SEtnbkhFLQJya4ZoG7jq6c"; // key to access API
+
+  public static final HttpTransport HTTP_TRANSPORT = new NetHttpTransport();
+  public static final JsonFactory JSON_FACTORY = new JacksonFactory();
+  private static final long NUMBER_OF_VIDEOS_RETURNED = 6;
+
+  private static final ArrayList<YTVid> EMPTY_LIST = new ArrayList<>();
+
+  /** Initialize a YouTube object to search for videos on YouTube */
+  private static YouTube youtube;
+
+  /** Initialize a YouTube object to search for videos on YouTube */
+  public static ArrayList<YTVid> getData(String regionCode) {
+    Properties properties = new Properties();
+    try {
+      // Object to make YouTube Data API requests
+      youtube =
+          new YouTube.Builder(
+                  HTTP_TRANSPORT,
+                  JSON_FACTORY,
+                  new HttpRequestInitializer() {
+                    public void initialize(HttpRequest request) throws IOException {}
+                  })
+              .setApplicationName("search-by-location")
+              .build();
+
+      // Define the API request for retrieving result.
+      YouTube.Videos.List search = youtube.videos().list("snippet,contentDetails,statistics");
+
+      search.setKey(apiKey);
+      search.setMaxResults(NUMBER_OF_VIDEOS_RETURNED);
+      search.setRegionCode(regionCode);
+      search.setChart("mostPopular");
+
+      // Call the API and return results.
+      VideoListResponse searchResponse = search.execute();
+      List<Video> searchResultList = searchResponse.getItems();
+
+      if (searchResultList != null) {
+        ArrayList result = convertToYTVid(searchResultList.iterator());
+        return result;
+      }
+    } catch (GoogleJsonResponseException e) {
+      System.err.println(
+          "There was a service error: "
+              + e.getDetails().getCode()
+              + " : "
+              + e.getDetails().getMessage());
+    } catch (IOException e) {
+      System.err.println("There was an IO error: " + e.getCause() + " : " + e.getMessage());
+    } catch (Throwable t) {
+      t.printStackTrace();
+    }
+    return EMPTY_LIST;
+  }
+
+  /*
+   * function that creates YTVid file from singleVideo object
+   */
+  private static ArrayList<YTVid> convertToYTVid(Iterator<Video> iteratorSearchResults) {
+    ArrayList result = new ArrayList<>();
+    while (iteratorSearchResults.hasNext()) {
+      Video singleVideo = iteratorSearchResults.next();
+      String Id = singleVideo.getId();
+      String link = "https://www.youtube.com/watch?v=" + Id;
```

Comment:
It would be better to put this logic in the YTVid class and then construct YTVid using only the id and build the link in the constructor.
This would be more robust since the link creation would be done in a single place. Right now, it has to be done any time someone instantiates this class.